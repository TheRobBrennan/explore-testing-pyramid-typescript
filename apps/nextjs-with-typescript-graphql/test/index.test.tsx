import { InMemoryCache, gql } from "@apollo/client";
import React from "react";
import Index from "../pages";
import renderer, { act } from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";

// Refactor cache example to a shared resource
const cache = new InMemoryCache();
cache.writeQuery({
  query: gql`
    query Viewer {
      viewer {
        id
        name
        status
      }
    }
  `,
  data: {
    viewer: {
      __typename: "User",
      id: "Baa",
      name: "Baa",
      status: "Healthy",
    },
  },
});

describe("Page - Index [DEFAULT]", () => {
  it("matches the initially rendered snapshot", () => {
    const component = renderer.create(
      <MockedProvider cache={cache}>
        <Index />
      </MockedProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("matches the fully rendered snapshot", async () => {
    const component = renderer.create(
      <MockedProvider cache={cache}>
        <Index />
      </MockedProvider>
    );

    // Fire off async events to cause the component to change here
    await act(() => {
      return new Promise((resolve) => {
        // Simulate waiting for the GraphQL query response to populate our cache by waiting 10ms
        setTimeout(() => resolve(), 10);
      });
    });

    // Test the final state of our component
    expect(component.toJSON()).toMatchSnapshot();
  });
});
