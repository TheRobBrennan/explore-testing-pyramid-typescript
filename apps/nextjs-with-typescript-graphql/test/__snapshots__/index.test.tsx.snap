// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - Index [DEFAULT] matches the fully rendered snapshot 1`] = `
<div>
  You're signed in as 
  Baa
   and you're 
  Healthy
  . Go to the
   
  <a
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    about
  </a>
   
  page.
  <div>
    <input
      onChange={[Function]}
      placeholder="your new name..."
      type="text"
    />
    <input
      onClick={[Function]}
      type="button"
      value="change"
    />
  </div>
</div>
`;

exports[`Page - Index [DEFAULT] matches the initially rendered snapshot 1`] = `
<div>
  You're signed in as 
  Baa
   and you're 
  Healthy
  . Go to the
   
  <a
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    about
  </a>
   
  page.
  <div>
    <input
      onChange={[Function]}
      placeholder="your new name..."
      type="text"
    />
    <input
      onClick={[Function]}
      type="button"
      value="change"
    />
  </div>
</div>
`;
